src/app/modules/category/components/filter/filter.component.ts

```diff
@@ -1,12 +1,202 @@
-import { Component, OnInit } from '@angular/core';
+import {
+    Component,
+    OnDestroy,
+    OnInit,
+    ChangeDetectionStrategy,
+} from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Subject } from 'rxjs';
+import { filter, map, take, takeUntil, withLatestFrom } from 'rxjs/operators';
+import { ShareService } from 'src/app/services/share/share.service';
+import { FilterItem, FiltersData } from '../../state/category.models';
+import * as CategorySelectors from '../../state/selectors/category.selectors';
 
 @Component({
     selector: 'app-filter',
     templateUrl: './filter.component.html',
     styleUrls: ['./filter.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class FilterComponent implements OnInit {
-    constructor() {}
+export class FilterComponent implements OnInit, OnDestroy {
+    private readonly unsubscribe$ = new Subject();
+    readonly selectedItemsArray$ = this.shareService.getData();
+    private filterItemFromTo: FilterItem;
 
-    ngOnInit(): void {}
+    readonly filterItemsData$ = this.store
+        .select(CategorySelectors.selectFilterData)
+        .pipe(
+            filter(el => el !== undefined),
+            map(el => el),
+            takeUntil(this.unsubscribe$),
+        );
+
+    readonly isReadyToDisplay$ = this.filterItemsData$.pipe(map(el => !!el));
+
+    constructor(
+        private readonly store: Store,
+        private readonly shareService: ShareService,
+    ) {}
+
+    ngOnInit(): void {
+        this.shareService
+            .getData()
+            .pipe()
+            .subscribe(data => {
+                if (data === null) {
+                    data = [];
+                    this.shareService
+                        .getDeleteItem()
+                        .subscribe((item: FilterItem) => {
+                            const itemIndex = data.indexOf(item);
+                            data.splice(itemIndex, 1);
+                            this.shareService.sendData(data);
+                        });
+                }
+            });
+    }
+
+    onChecked(value: string, selected: boolean, id: number): void {
+        const filterItem = {
+            first_filter_key: value,
+            id,
+        };
+
+        this.selectedItemsArray$
+            .pipe(take(1), withLatestFrom(this.shareService.getData()))
```

Comment:
i can't get it, 
this.selectedItemsArray$ = this.shareService.getData();
and you use 
withLatestFrom(this.shareService.getData())
for what? it shoud be same streams...